xray/plot/plot.py

```diff
@@ -126,6 +132,18 @@ def plot(darray, ax=None, rtol=0.01, **kwargs):
     """
     darray = darray.squeeze()
 
+    if row or col:
+        allargs = locals().copy()
+        allargs.update(kwargs)
+
+        del allargs['kwargs']
+        del allargs['rtol']
+
+        # Assume that it's going to be the right shape and just do imshow.
+        # Can't use plot since it must have 2d plotting signature for
+        # map_dataarray
```

Comment:
I wonder if we could simply dispatch to the plotting submethods here, and keep all this logic there? That way, we'll still try to make a plot with the right number of dimensions, and the error gets raised there instead.